docker-compose.yaml

```diff
@@ -3,44 +3,49 @@ version: '3.8'
 services:
 
   server:
-    depends_on: [cache, ui]
+    depends_on:
+      cache:
+        condition: service_healthy
+      ui:
+        condition: service_healthy
+
```

Comment:
Added additional specifics that should help fix potential race conditions that could affect us if the services take some additional time to spin up.